cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,337 @@
+	let mut total_deposit = 0_u128;
+
+	// Set identity and Subs on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+
+		// 1. Set identity on Relay Chain
+		assert_ok!(RococoIdentity::set_identity(
+			RococoOrigin::signed(RococoRelaySender::get()),
+			Box::new(id.relay.clone())
+		));
+
+		// 2. Set sub-identity on Relay Chain
+		assert_ok!(RococoIdentity::set_subs(
+			RococoOrigin::signed(RococoRelaySender::get()),
+			vec![(RococoRelayReceiver::get(), Data::Raw(vec![42; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Identity(IdentityEvent::IdentitySet { .. }) => {},
+				RuntimeEvent::Balances(BalancesEvent::Reserved { .. }) => {},
+			]
+		);
+
+		let reserved_bal = RococoBalances::reserved_balance(RococoRelaySender::get());
+		total_deposit = SubAccountDeposit::get() + id_deposit_relaychain(&id.relay);
+
+		// The reserved balance should equal the calculated total deposit
+		assert_eq!(reserved_bal, total_deposit);
+	});
+	total_deposit
+}
+
+fn assert_set_id_parachain(id: &Identity) {
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		let free_bal = PeopleRococoBalances::free_balance(PeopleRococoSender::get());
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		// 3. Set identity on Parachain
+		assert_ok!(PeopleRococoIdentity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			id.para.clone(),
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(PeopleRococoIdentity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
+		// No events get triggered when calling set_sub_no_deposit
+
+		// No amount should be reserved as deposit amounts are set to 0.
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+		assert_eq!(reserved_bal, 0);
+		assert!(PeopleRococoIdentity::identity(&PeopleRococoSender::get()).is_some());
+		let (_, sub_accounts) =
+			<PeopleRococo as PeopleRococoPallet>::Identity::subs_of(&PeopleRococoSender::get());
+		assert_eq!(sub_accounts.len(), 1);
+	});
+}
+
+fn assert_reap_id_relay(total_deposit: u128, id: &Identity) {
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		let free_bal_before_reap = RococoBalances::free_balance(RococoRelaySender::get());
+		let reserved_balance = RococoBalances::reserved_balance(RococoRelaySender::get());
+		//before reap reserved balance should be equal to total deposit
+		assert_eq!(reserved_balance, total_deposit);
+
+		assert_ok!(RococoIdentityMigrator::reap_identity(
+			RococoOrigin::root(),
+			RococoRelaySender::get()
+		));
+
+		let remote_deposit = calculate_remote_deposit(id.relay.encoded_size() as u32, 1);
```

Comment:
Good point with the `1`, probably worthwhile to add some cases with 0 and `MaxSubAccounts`.

Reply:
I added those cases. Interestingly the case : `on_reap_indentity_works_for_full_identity_with_zero_subs()` is failing. We are not getting the expected event: `polkadot_runtime_common::identity_migrator::Event::DepositUpdated`. 

I would expect that even with no subs, this event should still be executed. 

The failing test is in the last commit 
`51a21a718eb92229bce36ad764521fe980c1b3e9`